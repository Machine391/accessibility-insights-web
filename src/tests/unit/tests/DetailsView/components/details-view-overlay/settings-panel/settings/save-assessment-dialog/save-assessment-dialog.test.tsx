// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { IMock, Mock } from 'typemoq';
import {
    SaveAssessmentDialog,
    SaveAssessmentDialogProps,
} from '../../../../../../../../../DetailsView/components/save-assessment-dialog';
import { AssessmentActionMessageCreator } from '../../../../../../../../../DetailsView/actions/assessment-action-message-creator';
import { UserConfigMessageCreator } from '../../../../../../../../../common/message-creators/user-config-message-creator';
import { UserConfigurationStoreData } from '../../../../../../../../../common/types/store-data/user-configuration-store';
import { render } from '@testing-library/react';
import React from 'react';

describe('SaveAssessmentButton', () => {
    it('snapshot of dialog', () => {
        let propsStub: SaveAssessmentDialogProps;
        let assessmentActionMessageCreatorMock: IMock<AssessmentActionMessageCreator>;
        let userConfigMessageCreatorMock: IMock<UserConfigMessageCreator>;
        let userConfigurationStoreData: UserConfigurationStoreData;
        assessmentActionMessageCreatorMock = Mock.ofType<AssessmentActionMessageCreator>();
        userConfigurationStoreData = {
            showSaveAssessmentDialog: true,
        } as UserConfigurationStoreData;
        userConfigMessageCreatorMock = Mock.ofType(UserConfigMessageCreator);
        propsStub = {
            deps: {
                getAssessmentActionMessageCreator: () => assessmentActionMessageCreatorMock.object,
                userConfigMessageCreator: userConfigMessageCreatorMock.object,
            },
            userConfigurationStoreData,
            isOpen: true,
            onClose: () => !userConfigurationStoreData.showSaveAssessmentDialog,
        };
        const wrapper = render(<SaveAssessmentDialog {...propsStub} />);
        expect(wrapper.asFragment()).toMatchSnapshot();
    });
});
