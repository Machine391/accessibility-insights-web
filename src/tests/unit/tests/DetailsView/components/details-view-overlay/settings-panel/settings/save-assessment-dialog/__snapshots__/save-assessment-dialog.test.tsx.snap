// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveAssessmentButton snapshot of dialog 1`] = `
<DocumentFragment>
  <span
    class="ms-layer"
  />
</DocumentFragment>
`;
